src/Illuminate/Database/Schema/Grammars/MySqlGrammar.php

```diff
@@ -43,11 +43,13 @@ class MySqlGrammar extends Grammar
      */
     public function compileCreateDatabase($name, $connection)
     {
+        $collation = $connection->getConfig('collation');
+
         return sprintf(
             'create database %s default character set %s default collate %s',
```

Comment:
@Jubeki can you confirm this can generate the correct SQL?

Reply:
Seems like for Mysql this results in the following:

```php
$grammar = new MySqlGrammar();
return $grammar->compileCreateDatabase('test', DB::connection());

// create database `test` default character set `utf8mb4` default collate
```

Result of execution of the sql query:

```
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1
```

(I will also take a look at the linked issue later today)

Reply:
I tested the current `laravel/laravel` against Laravel Sail, I don't have any other DB on may Laptop.

Simplest solution would probably be to revert my PR @crynobone, and simply add another config entry for Laravel 11. (see my framework PR)

Reply:
@Jubeki just something to consider for Laravel 11, `Schema::createDatabase()` is used in Parallel testing and similar issue will still persist in Laravel 11 if `collation` configuration is `null`.

Reply:
Yeah, for Laravel 11 was the idea to create the new mariadb config entry with the default collation set for both mariadb and mysql. (see https://github.com/laravel/framework/pull/48455)